src/marks/auto.js

```diff
@@ -0,0 +1,269 @@
+import {InternSet} from "d3";
+import {isOrdinal, labelof, valueof, isOptions, identity, isColor} from "../options.js";
+import {area, areaX, areaY} from "./area.js";
+import {dot} from "./dot.js";
+import {line, lineX, lineY} from "./line.js";
+import {ruleX, ruleY} from "./rule.js";
+import {barX, barY} from "./bar.js";
+import {rect, rectX, rectY} from "./rect.js";
+import {cell} from "./cell.js";
+import {frame} from "./frame.js";
+import {bin, binX, binY} from "../transforms/bin.js";
+import {group, groupX, groupY} from "../transforms/group.js";
+import {marks} from "../mark.js";
+import {ascending} from "d3";
+
+export function auto(data, {x, y, fx, fy, color, size, mark} = {}) {
+  // Shorthand: array of primitives should result in a histogram
+  if (x === undefined && y === undefined) x = identity;
```

Comment:
Last blocker, IMO: I think this line for the "shorthand" is now _too_ permissive. We probably want to check that the first element is a primitive, or else throw an error if no options are specified? Currently `Plot.auto(aapl).plot()` looks like this:

<img width="652" alt="image" src="https://user-images.githubusercontent.com/841829/216516199-d5bc2088-787e-45e5-b9af-0a1e64e3f3f0.png">

Reply:
I suggest we drop this shorthand entirely for now. Maybe Plot.auto returns null or throws an error? I dunno.

Reply:
Yeah probably. If I just remove the line it looks like this:

<img width="928" alt="image" src="https://user-images.githubusercontent.com/841829/216520584-4072c607-b04a-4aee-acda-78ca4dc3c67d.png">

Which I think looks broken. So I added an error:

<img width="268" alt="image" src="https://user-images.githubusercontent.com/841829/216521741-a757dd36-9670-4284-af2c-fe739065d370.png">